### ISSUE TITLE
Cryptic "TearDown : System.InvalidOperationException : TestThreadContext with id  was already released" Error after Migrating from Specflow

### ISSUE THREAD

**Issue by thatstatsguy** [reqnroll/Reqnroll#295]
### Reqnroll Version

2.1.1

### Which test runner are you using?

NUnit

### Test Runner Version Number

2.1.1

### .NET Implementation

.NET 8.0

### Test Execution Method

ReSharper Test Runner

### Content of reqnroll.json configuration file

N/A

### Issue Description

Hi :) 

This is more of a "We experience this - here's the fix - we don't know why it happened"

I recently moved over from Specflow to ReqnRoll. Migration went pretty smoothly. Only hitch we ran into was the following error message when running our end to end tests.

```
TearDown : System.InvalidOperationException : TestThreadContext with id  was already released
```

Additional note: All code/tests executed successfully on specflow prior to migrating to reqnroll

### Steps to Reproduce

To reproduce one can simply run the following pre test hook
```
[BeforeTestRun]
    public static async Task LoginUser(TestThreadContext testThreadContext)
    {
            // .... other steps above redacted ....
            var email = new MailAddress("foo@gmail.com");
            testThreadContext["Email"] = email ;
             // .... other steps belowredacted ....
    }
```

Along with a post test hook

```
[AfterTestRun]
    public static async Task LogoutUser()
    {
        //do literally anything here, just needs to execute
    }
```

running this with a simple tests reproduced the error for me.

We just so happened to not actually need to use the test thread context and could safely remove it from the pre test run hook (it was old code that was used to test an idea but we never actually used it anywhere) i.e. 
```
[BeforeTestRun]
    public static async Task LoginUser()
    {
            // no code here relies on test threadcontext
    }
```

**Summary**
I don't know a huge amount about how the test thread context is intended to be used but my suspicion is that we were using it incorrectly. Either way - thought it posting this for awareness for those who run into this while migrating!

**Comment by gasparnagy** [comment]
Thanks for posting the issue. 

It is a recurring problem (duplicate of #247). In the Before and After Test Run hooks you cannot use the Test Thread context. (There are anyway multiple Test Thread context if you run tests parallel so you cannot initialize it in the Before/After Test Run hook.)

Unfortunately SpecFlow had a bug and executed the Before/After Test Run hook in the wrong context (test-thread context instead of the global context). This is why the code was "working" for you. But we needed to fix that bug for other reasons, so this code is not working anymore. 

In #247 I posted workarounds for it.

I close this issue as a duplicate for now.
